// ng2-date-picker/dp-month-calendar/dp-month-calendar.component.js
"use strict";
var core_1 = require("@angular/core");
var month_calendar_service_1 = require("./month-calendar.service");
var MonthCalendarComponent = (function () {
    function MonthCalendarComponent(monthCalendarService) {
        this.monthCalendarService = monthCalendarService;
        this.monthClick = new core_1.EventEmitter();
    }
    MonthCalendarComponent.prototype.ngOnInit = function () {
        this.yearMonths = this.monthCalendarService.generateYear(this.config.month, this.selected);
    };
    MonthCalendarComponent.prototype.isDisabledMonth = function (month) {
        return this.monthCalendarService.isDateDisabled(month, this.config);
    };
    MonthCalendarComponent.prototype.monthSelected = function (month) {
        this.monthClick.emit(month.date);
    };
    return MonthCalendarComponent;
}());
MonthCalendarComponent.decorators = [
    { type: core_1.Component, args: [{
                selector: 'dp-month-calendar',
                template: '<div class="dp-calendar-wrapper">   <div class="dp-months-row" *ngFor="let monthRow of yearMonths">     <button class="dp-calendar-month"             *ngFor="let month of monthRow"             [disabled]="isDisabledMonth(month)"             [ngClass]="{\'dp-selected\': month.selected,\'dp-current-month\': month.currentMonth}"             (click)="monthSelected(month)">       {{month.date.format(\'MMM\')}}     </button>   </div> </div>',
                styles: ['.dp-calendar-month {  box-sizing: border-box;  width: 52.5px;  height: 52.5px;  cursor: pointer;}.dp-calendar-month.dp-selected {  background: #106CC8;  color: #FFFFFF;}'],
                providers: [month_calendar_service_1.MonthCalendarService]
            },] },
];
/** @nocollapse */
MonthCalendarComponent.ctorParameters = function () { return [
    { type: month_calendar_service_1.MonthCalendarService, },
]; };
MonthCalendarComponent.propDecorators = {
    'config': [{ type: core_1.Input },],
    'selected': [{ type: core_1.Input },],
    'monthClick': [{ type: core_1.Output },],
};
exports.MonthCalendarComponent = MonthCalendarComponent;
