// ng2-date-picker/dp-month-calendar/month-calendar.service.js
"use strict";
var core_1 = require("@angular/core");
var moment = require("moment");
var utils_service_1 = require("../common/services/utils/utils.service");
var MonthCalendarService = (function () {
    function MonthCalendarService(utilsService) {
        this.utilsService = utilsService;
    }
    MonthCalendarService.prototype.generateYear = function (year, selected) {
        var _this = this;
        if (selected === void 0) { selected = null; }
        var index = year.clone().startOf('year');
        return this.utilsService.createArray(3).map(function () {
            return _this.utilsService.createArray(4).map(function () {
                var month = {
                    date: index.clone(),
                    selected: index.isSame(selected, 'month'),
                    currentMonth: index.isSame(moment(), 'month')
                };
                index.add(1, 'month');
                return month;
            });
        });
    };
    MonthCalendarService.prototype.isDateDisabled = function (month, config) {
        if (config.min && month.date.isBefore(config.min, 'month')) {
            return true;
        }
        return !!(config.max && month.date.isAfter(config.max, 'month'));
    };
    return MonthCalendarService;
}());
MonthCalendarService.decorators = [
    { type: core_1.Injectable },
];
/** @nocollapse */
MonthCalendarService.ctorParameters = function () { return [
    { type: utils_service_1.UtilsService, },
]; };
exports.MonthCalendarService = MonthCalendarService;
