// @ngrx/store-log-monitor/src/json-tree/types.js
import { compose } from '@ngrx/core/compose';
export var KNOWN = {
    Array: 'array',
    Object: 'object',
    Null: 'null',
    Undefined: 'undefined',
    Boolean: 'boolean',
    Number: 'number',
    String: 'string',
    Symbol: 'symbol',
    Function: 'function',
    Iterable: 'iterable'
};
export function getTypeOf(object) {
    var literalType = typeof object;
    if (literalType === 'object') {
        if (Array.isArray(object)) {
            return KNOWN.Array;
        }
        if (object === null) {
            return KNOWN.Null;
        }
        if (typeof object[Symbol.iterator] === 'function') {
            return KNOWN.Iterable;
        }
    }
    return literalType;
}
var arrayLength = function (value) { return value.length; };
var lengthLabel = function (single, plural) { return function (length) { return (length + " " + (length === 1 ? single : plural)); }; };
var typeIndicator = function (typeIndicator) { return function (input) { return (typeIndicator + " " + input); }; };
var typeIdentity = function (type) { return function () { return type; }; };
var withQuotes = function (val) { return ("\"" + val + "\""); };
var toString = function (val) { return val.toString(); };
var iterableToArray = function (value) { return Array.from(value); };
var labelFactoriesForTypes = (_a = {},
    _a[KNOWN.Array] = compose(typeIndicator('[]'), lengthLabel('item', 'items'), arrayLength),
    _a[KNOWN.Object] = compose(typeIndicator('{}'), lengthLabel('key', 'keys'), arrayLength, Object.getOwnPropertyNames),
    _a[KNOWN.Null] = typeIdentity(KNOWN.Null),
    _a[KNOWN.Undefined] = typeIdentity(KNOWN.Undefined),
    _a[KNOWN.Boolean] = function (val) { return val ? 'true' : 'false'; },
    _a[KNOWN.Number] = toString,
    _a[KNOWN.String] = withQuotes,
    _a[KNOWN.Symbol] = compose(withQuotes, toString),
    _a[KNOWN.Function] = typeIdentity(KNOWN.Function),
    _a[KNOWN.Iterable] = compose(typeIndicator('()'), lengthLabel('entry', 'entries'), arrayLength, iterableToArray),
    _a
);
var lookupLabelForType = function (type) { return labelFactoriesForTypes[type]; };
export var getLabelFor = function (object) { return labelFactoriesForTypes[getTypeOf(object)](object); };
export function getChildrenFor(object) {
    var literalType = getTypeOf(object);
    if (literalType === KNOWN.Object) {
        return Object.getOwnPropertyNames(object).map(function (name) {
            return { key: name, value: object[name] };
        });
    }
    else if (literalType === KNOWN.Array) {
        return object.map(function (value, index) {
            return { key: index, value: value };
        });
    }
    else if (literalType === KNOWN.Iterable) {
        return Array.from(object).map(function (value, index) {
            return { key: index, value: value };
        });
    }
    throw new TypeError("Tried to get children for non-enumerable type \"" + literalType + "\"");
}
var _a;
