// @ngrx/store-log-monitor/src/dock-monitor/index.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { CommanderComponent } from './commander';
import { DockComponent } from './dock';
import { DockMonitorComponent } from './dock-monitor';
import { DockActions } from './actions';
export var DockMonitorModule = (function () {
    function DockMonitorModule() {
    }
    DockMonitorModule.decorators = [
        { type: NgModule, args: [{
                    imports: [
                        CommonModule
                    ],
                    declarations: [
                        CommanderComponent,
                        DockComponent,
                        DockMonitorComponent
                    ],
                    providers: [
                        DockActions
                    ],
                    exports: [
                        DockMonitorComponent
                    ]
                },] },
    ];
    /** @nocollapse */
    DockMonitorModule.ctorParameters = [];
    return DockMonitorModule;
}());
export { useDockMonitor } from './reducer';
