// ng2-date-picker/dp-day-calendar/dp-day-calendar.component.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular 2 template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */

import * as import0 from 'ng2-date-picker/dp-day-calendar/dp-day-calendar.component';
import * as import1 from '@angular/core/src/change_detection/change_detection_util';
import * as import2 from '@angular/core/src/linker/view';
import * as import3 from '@angular/core/src/linker/view_utils';
import * as import4 from '@angular/core/src/linker/debug_context';
import * as import5 from 'ng2-date-picker/dp-day-calendar/day-calendar.service';
import * as import6 from '@angular/core/src/render/api';
import * as import7 from '@angular/core/src/metadata/view';
import * as import8 from '@angular/core/src/linker/view_type';
import * as import9 from '@angular/core/src/change_detection/constants';
import * as import10 from '@angular/core/src/linker/component_factory';
import * as import11 from 'ng2-date-picker/common/services/utils/utils.service';
import * as import12 from '@angular/core/src/linker/view_container';
import * as import13 from '@angular/common/src/directives/ng_class';
import * as import14 from '../../@angular/common/src/directives/ng_class.ngfactory';
import * as import15 from '@angular/core/src/change_detection/differs/iterable_differs';
import * as import16 from '@angular/core/src/change_detection/differs/keyvalue_differs';
import * as import17 from '@angular/core/src/linker/element_ref';
import * as import18 from '@angular/core/src/linker/template_ref';
import * as import19 from '@angular/common/src/directives/ng_if';
import * as import20 from '@angular/common/src/directives/ng_for';
import * as import21 from '../../@angular/common/src/directives/ng_if.ngfactory';
import * as import22 from '../../@angular/common/src/directives/ng_for.ngfactory';
export class Wrapper_DayCalendarComponent {
  /*private*/ _eventHandler:Function;
  context:import0.DayCalendarComponent;
  /*private*/ _changed:boolean;
  /*private*/ _changes:{[key: string]:any};
  /*private*/ _expr_0:any;
  /*private*/ _expr_1:any;
  subscription0:any;
  subscription1:any;
  constructor(p0:any) {
    this._changed = false;
    this._changes = {};
    this.context = new import0.DayCalendarComponent(p0);
    this._expr_0 = import1.UNINITIALIZED;
    this._expr_1 = import1.UNINITIALIZED;
  }
  ngOnDetach(view:import2.AppView<any>,componentView:import2.AppView<any>,el:any):void {
  }
  ngOnDestroy():void {
    (this.subscription0 && this.subscription0.unsubscribe());
    (this.subscription1 && this.subscription1.unsubscribe());
  }
  check_config(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import3.checkBinding(throwOnChange,this._expr_0,currValue))) {
      this._changed = true;
      this.context.config = currValue;
      this._changes['config'] = new import1.SimpleChange(this._expr_0,currValue);
      this._expr_0 = currValue;
    }
  }
  check_selected(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import3.checkBinding(throwOnChange,this._expr_1,currValue))) {
      this._changed = true;
      this.context.selected = currValue;
      this._changes['selected'] = new import1.SimpleChange(this._expr_1,currValue);
      this._expr_1 = currValue;
    }
  }
  ngDoCheck(view:import2.AppView<any>,el:any,throwOnChange:boolean):boolean {
    var changed:any = this._changed;
    this._changed = false;
    if (!throwOnChange) {
      if (changed) {
        this.context.ngOnChanges(this._changes);
        this._changes = {};
      }
      if ((view.numberOfChecks === 0)) { this.context.ngOnInit(); }
    }
    return changed;
  }
  checkHost(view:import2.AppView<any>,componentView:import2.AppView<any>,el:any,throwOnChange:boolean):void {
  }
  handleEvent(eventName:string,$event:any):boolean {
    var result:boolean = true;
    return result;
  }
  subscribe(view:import2.AppView<any>,_eventHandler:any,emit0:boolean,emit1:boolean):void {
    this._eventHandler = _eventHandler;
    if (emit0) { (this.subscription0 = this.context.dayClick.subscribe(_eventHandler.bind(view,'dayClick'))); }
    if (emit1) { (this.subscription1 = this.context.dayContextMenu.subscribe(_eventHandler.bind(view,'dayContextMenu'))); }
  }
}
  const nodeDebugInfos_DayCalendarComponent_Host0:import4.StaticNodeDebugInfo[] = [new import4.StaticNodeDebugInfo([
    import5.CalendarMonthService,
    import0.DayCalendarComponent
  ]
,import0.DayCalendarComponent,{})];
var renderType_DayCalendarComponent_Host:import6.RenderComponentType = import3.createRenderComponentType('',0,import7.ViewEncapsulation.None,([] as any[]),{});
class View_DayCalendarComponent_Host0 extends import2.DebugAppView<any> {
  _el_0:any;
  compView_0:import2.AppView<import0.DayCalendarComponent>;
  _CalendarMonthService_0_3:import5.CalendarMonthService;
  _DayCalendarComponent_0_4:Wrapper_DayCalendarComponent;
  constructor(viewUtils:import3.ViewUtils,parentView:import2.AppView<any>,parentIndex:number,parentElement:any) {
    super(View_DayCalendarComponent_Host0,renderType_DayCalendarComponent_Host,import8.ViewType.HOST,viewUtils,parentView,parentIndex,parentElement,import9.ChangeDetectorStatus.CheckAlways,nodeDebugInfos_DayCalendarComponent_Host0);
  }
  createInternal(rootSelector:string):import10.ComponentRef<any> {
    this._el_0 = import3.selectOrCreateRenderHostElement(this.renderer,'dp-day-calendar',import3.EMPTY_INLINE_ARRAY,rootSelector,this.debug(0,0,0));
    this.compView_0 = new View_DayCalendarComponent0(this.viewUtils,this,0,this._el_0);
    this._CalendarMonthService_0_3 = new import5.CalendarMonthService(this.injectorGet(import11.UtilsService,this.parentIndex));
    this._DayCalendarComponent_0_4 = new Wrapper_DayCalendarComponent(this._CalendarMonthService_0_3);
    this.compView_0.create(this._DayCalendarComponent_0_4.context);
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [this._el_0]),(null as any));
    return new import10.ComponentRef_<any>(0,this,this._el_0,this._DayCalendarComponent_0_4.context);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import5.CalendarMonthService) && (0 === requestNodeIndex))) { return this._CalendarMonthService_0_3; }
    if (((token === import0.DayCalendarComponent) && (0 === requestNodeIndex))) { return this._DayCalendarComponent_0_4.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    this.debug(0,0,0);
    this._DayCalendarComponent_0_4.ngDoCheck(this,this._el_0,throwOnChange);
    this.compView_0.internalDetectChanges(throwOnChange);
  }
  destroyInternal():void {
    this.compView_0.destroy();
    this._DayCalendarComponent_0_4.ngOnDestroy();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
}
export const DayCalendarComponentNgFactory:import10.ComponentFactory<import0.DayCalendarComponent> = new import10.ComponentFactory<import0.DayCalendarComponent>('dp-day-calendar',View_DayCalendarComponent_Host0,import0.DayCalendarComponent);
const styles_DayCalendarComponent:any[] = ['[_nghost-%COMP%] {  display: inline-block;}.dp-calendar-wrapper[_ngcontent-%COMP%] {  box-sizing: border-box;  border: 1px solid #000000;}.dp-calendar-wrapper[_ngcontent-%COMP%]   .dp-calendar-weekday[_ngcontent-%COMP%]:first-child {  border-left: none;}.dp-calendar-weekday[_ngcontent-%COMP%] {  box-sizing: border-box;  display: inline-block;  width: 30px;  text-align: center;  border-left: 1px solid #000000;  border-bottom: 1px solid #000000;}.dp-calendar-day[_ngcontent-%COMP%] {  box-sizing: border-box;  width: 30px;  height: 30px;  cursor: pointer;}.dp-selected[_ngcontent-%COMP%] {  background: #106CC8;  color: #FFFFFF;}.dp-prev-month[_ngcontent-%COMP%], .dp-next-month[_ngcontent-%COMP%] {  opacity: 0.5;}.dp-hide-near-month[_ngcontent-%COMP%]   .dp-prev-month[_ngcontent-%COMP%], .dp-hide-near-month[_ngcontent-%COMP%]   .dp-next-month[_ngcontent-%COMP%] {  visibility: hidden;}.dp-week-number[_ngcontent-%COMP%] {  position: absolute;  font-size: 9px;}'];
const nodeDebugInfos_DayCalendarComponent1:import4.StaticNodeDebugInfo[] = [
  new import4.StaticNodeDebugInfo(([] as any[]),(null as any),{}),
  new import4.StaticNodeDebugInfo(([] as any[]),(null as any),{})
]
;
class View_DayCalendarComponent1 extends import2.DebugAppView<any> {
  _el_0:any;
  _text_1:any;
  /*private*/ _expr_2:any;
  constructor(viewUtils:import3.ViewUtils,parentView:import2.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import12.ViewContainer) {
    super(View_DayCalendarComponent1,renderType_DayCalendarComponent,import8.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import9.ChangeDetectorStatus.CheckAlways,nodeDebugInfos_DayCalendarComponent1,declaredViewContainer);
    this._expr_2 = import1.UNINITIALIZED;
  }
  createInternal(rootSelector:string):import10.ComponentRef<any> {
    this._el_0 = import3.createRenderElement(this.renderer,(null as any),'span',new import3.InlineArray2(2,'class','dp-calendar-weekday'),this.debug(0,0,128));
    this._text_1 = this.renderer.createText(this._el_0,'',this.debug(1,0,195));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1
    ]
    ),(null as any));
    return (null as any);
  }
  detectChangesInternal(throwOnChange:boolean):void {
    this.debug(1,0,195);
    const currVal_2:any = import3.inlineInterpolate(1,'',this.context.$implicit,'');
    if (import3.checkBinding(throwOnChange,this._expr_2,currVal_2)) {
      this.renderer.setText(this._text_1,currVal_2);
      this._expr_2 = currVal_2;
    }
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
}
const nodeDebugInfos_DayCalendarComponent3:import4.StaticNodeDebugInfo[] = [
  new import4.StaticNodeDebugInfo(([] as any[]),(null as any),{}),
  new import4.StaticNodeDebugInfo(([] as any[]),(null as any),{})
]
;
class View_DayCalendarComponent3 extends import2.DebugAppView<any> {
  _el_0:any;
  _text_1:any;
  /*private*/ _expr_2:any;
  constructor(viewUtils:import3.ViewUtils,parentView:import2.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import12.ViewContainer) {
    super(View_DayCalendarComponent3,renderType_DayCalendarComponent,import8.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import9.ChangeDetectorStatus.CheckAlways,nodeDebugInfos_DayCalendarComponent3,declaredViewContainer);
    this._expr_2 = import1.UNINITIALIZED;
  }
  createInternal(rootSelector:string):import10.ComponentRef<any> {
    this._el_0 = import3.createRenderElement(this.renderer,(null as any),'span',new import3.InlineArray2(2,'class','dp-week-number'),this.debug(0,0,287));
    this._text_1 = this.renderer.createText(this._el_0,'',this.debug(1,0,347));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1
    ]
    ),(null as any));
    return (null as any);
  }
  detectChangesInternal(throwOnChange:boolean):void {
    this.debug(1,0,347);
    const currVal_2:any = import3.inlineInterpolate(1,'',this.parentView.context.$implicit[0].date.isoWeek(),'');
    if (import3.checkBinding(throwOnChange,this._expr_2,currVal_2)) {
      this.renderer.setText(this._text_1,currVal_2);
      this._expr_2 = currVal_2;
    }
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
}
const nodeDebugInfos_DayCalendarComponent4:import4.StaticNodeDebugInfo[] = [
  new import4.StaticNodeDebugInfo([import13.NgClass],(null as any),{}),
  new import4.StaticNodeDebugInfo(([] as any[]),(null as any),{})
]
;
class View_DayCalendarComponent4 extends import2.DebugAppView<any> {
  _el_0:any;
  _NgClass_0_3:import14.Wrapper_NgClass;
  _text_1:any;
  /*private*/ _expr_3:any;
  _map_4:any;
  /*private*/ _expr_5:any;
  constructor(viewUtils:import3.ViewUtils,parentView:import2.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import12.ViewContainer) {
    super(View_DayCalendarComponent4,renderType_DayCalendarComponent,import8.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import9.ChangeDetectorStatus.CheckAlways,nodeDebugInfos_DayCalendarComponent4,declaredViewContainer);
    this._expr_3 = import1.UNINITIALIZED;
    this._map_4 = import3.pureProxy5((p0:any,p1:any,p2:any,p3:any,p4:any):{[key: string]:any} => {
      return {
        'dp-selected': p0,
        'dp-current-month': p1,
        'dp-prev-month': p2,
        'dp-next-month': p3,
        'dp-current-day': p4
      }
      ;
    });
    this._expr_5 = import1.UNINITIALIZED;
  }
  createInternal(rootSelector:string):import10.ComponentRef<any> {
    this._el_0 = import3.createRenderElement(this.renderer,(null as any),'button',new import3.InlineArray2(2,'class','dp-calendar-day'),this.debug(0,0,385));
    this._NgClass_0_3 = new import14.Wrapper_NgClass(this.parentView.parentView.parentView.injectorGet(import15.IterableDiffers,this.parentView.parentView.parentIndex),this.parentView.parentView.parentView.injectorGet(import16.KeyValueDiffers,this.parentView.parentView.parentIndex),new import17.ElementRef(this._el_0),this.renderer);
    this._text_1 = this.renderer.createText(this._el_0,'',this.debug(1,0,835));
    var disposable_0:Function = import3.subscribeToRenderElement(this,this._el_0,new import3.InlineArray2(2,'click',(null as any)),this.eventHandler(this.handleEvent_0));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1
    ]
    ),[disposable_0]);
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import13.NgClass) && ((0 <= requestNodeIndex) && (requestNodeIndex <= 1)))) { return this._NgClass_0_3.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    this.debug(0,0,393);
    const currVal_0_0_0:any = 'dp-calendar-day';
    this._NgClass_0_3.check_klass(currVal_0_0_0,throwOnChange,false);
    this.debug(0,0,573);
    const currVal_0_0_1:any = this._map_4(this.context.$implicit.selected,this.context.$implicit.currentMonth,this.context.$implicit.prevMonth,this.context.$implicit.nextMonth,this.context.$implicit.currentDay);
    this._NgClass_0_3.check_ngClass(currVal_0_0_1,throwOnChange,false);
    this._NgClass_0_3.ngDoCheck(this,this._el_0,throwOnChange);
    this.debug(0,0,529);
    const currVal_3:any = this.parentView.parentView.context.isDisabledDay(this.context.$implicit);
    if (import3.checkBinding(throwOnChange,this._expr_3,currVal_3)) {
      this.renderer.setElementProperty(this._el_0,'disabled',currVal_3);
      this._expr_3 = currVal_3;
    }
    this.debug(1,0,835);
    const currVal_5:any = import3.inlineInterpolate(1,'       ',this.context.$implicit.date.format('DD'),'     ');
    if (import3.checkBinding(throwOnChange,this._expr_5,currVal_5)) {
      this.renderer.setText(this._text_1,currVal_5);
      this._expr_5 = currVal_5;
    }
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
  handleEvent_0(eventName:string,$event:any):boolean {
    this.debug(0,0,385);
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    if ((eventName == 'click')) {
      const pd_sub_0:any = ((<any>this.parentView.parentView.context.dayClick.emit({
        day: this.context.$implicit,
        event: $event
      }
      )) !== false);
      result = (pd_sub_0 && result);
    }
    return result;
  }
}
const nodeDebugInfos_DayCalendarComponent2:import4.StaticNodeDebugInfo[] = [
  new import4.StaticNodeDebugInfo(([] as any[]),(null as any),{}),
  new import4.StaticNodeDebugInfo(([] as any[]),(null as any),{}),
  new import4.StaticNodeDebugInfo([
    import18.TemplateRef,
    import19.NgIf
  ]
  ,(null as any),{}),
  new import4.StaticNodeDebugInfo(([] as any[]),(null as any),{}),
  new import4.StaticNodeDebugInfo([
    import18.TemplateRef,
    import20.NgFor
  ]
  ,(null as any),{}),
  new import4.StaticNodeDebugInfo(([] as any[]),(null as any),{})
]
;
class View_DayCalendarComponent2 extends import2.DebugAppView<any> {
  _el_0:any;
  _text_1:any;
  _anchor_2:any;
  /*private*/ _vc_2:import12.ViewContainer;
  _TemplateRef_2_5:any;
  _NgIf_2_6:import21.Wrapper_NgIf;
  _text_3:any;
  _anchor_4:any;
  /*private*/ _vc_4:import12.ViewContainer;
  _TemplateRef_4_5:any;
  _NgFor_4_6:import22.Wrapper_NgFor;
  _text_5:any;
  constructor(viewUtils:import3.ViewUtils,parentView:import2.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import12.ViewContainer) {
    super(View_DayCalendarComponent2,renderType_DayCalendarComponent,import8.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import9.ChangeDetectorStatus.CheckAlways,nodeDebugInfos_DayCalendarComponent2,declaredViewContainer);
  }
  createInternal(rootSelector:string):import10.ComponentRef<any> {
    this._el_0 = import3.createRenderElement(this.renderer,(null as any),'div',new import3.InlineArray2(2,'class','dp-calendar-week'),this.debug(0,0,225));
    this._text_1 = this.renderer.createText(this._el_0,'     ',this.debug(1,0,282));
    this._anchor_2 = this.renderer.createTemplateAnchor(this._el_0,this.debug(2,0,287));
    this._vc_2 = new import12.ViewContainer(2,0,this,this._anchor_2);
    this._TemplateRef_2_5 = new import18.TemplateRef_(this,2,this._anchor_2);
    this._NgIf_2_6 = new import21.Wrapper_NgIf(this._vc_2.vcRef,this._TemplateRef_2_5);
    this._text_3 = this.renderer.createText(this._el_0,'     ',this.debug(3,0,380));
    this._anchor_4 = this.renderer.createTemplateAnchor(this._el_0,this.debug(4,0,385));
    this._vc_4 = new import12.ViewContainer(4,0,this,this._anchor_4);
    this._TemplateRef_4_5 = new import18.TemplateRef_(this,4,this._anchor_4);
    this._NgFor_4_6 = new import22.Wrapper_NgFor(this._vc_4.vcRef,this._TemplateRef_4_5,this.parentView.parentView.injectorGet(import15.IterableDiffers,this.parentView.parentIndex),this.parentView.ref);
    this._text_5 = this.renderer.createText(this._el_0,'   ',this.debug(5,0,881));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1,
      this._anchor_2,
      this._text_3,
      this._anchor_4,
      this._text_5
    ]
    ),(null as any));
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import18.TemplateRef) && (2 === requestNodeIndex))) { return this._TemplateRef_2_5; }
    if (((token === import19.NgIf) && (2 === requestNodeIndex))) { return this._NgIf_2_6.context; }
    if (((token === import18.TemplateRef) && (4 === requestNodeIndex))) { return this._TemplateRef_4_5; }
    if (((token === import20.NgFor) && (4 === requestNodeIndex))) { return this._NgFor_4_6.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    this.debug(2,0,293);
    const currVal_2_0_0:any = this.parentView.context.config.showWeekNumbers;
    this._NgIf_2_6.check_ngIf(currVal_2_0_0,throwOnChange,false);
    this._NgIf_2_6.ngDoCheck(this,this._anchor_2,throwOnChange);
    this.debug(4,0,429);
    const currVal_4_0_0:any = this.context.$implicit;
    this._NgFor_4_6.check_ngForOf(currVal_4_0_0,throwOnChange,false);
    this._NgFor_4_6.ngDoCheck(this,this._anchor_4,throwOnChange);
    this._vc_2.detectChangesInNestedViews(throwOnChange);
    this._vc_4.detectChangesInNestedViews(throwOnChange);
  }
  destroyInternal():void {
    this._vc_2.destroyNestedViews();
    this._vc_4.destroyNestedViews();
    this.debug(2,0,287);
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
  createEmbeddedViewInternal(nodeIndex:number):import2.AppView<any> {
    if ((nodeIndex == 2)) { return new View_DayCalendarComponent3(this.viewUtils,this,2,this._anchor_2,this._vc_2); }
    if ((nodeIndex == 4)) { return new View_DayCalendarComponent4(this.viewUtils,this,4,this._anchor_4,this._vc_4); }
    return (null as any);
  }
}
const nodeDebugInfos_DayCalendarComponent0:import4.StaticNodeDebugInfo[] = [
  new import4.StaticNodeDebugInfo([import13.NgClass],(null as any),{}),
  new import4.StaticNodeDebugInfo(([] as any[]),(null as any),{}),
  new import4.StaticNodeDebugInfo(([] as any[]),(null as any),{}),
  new import4.StaticNodeDebugInfo(([] as any[]),(null as any),{}),
  new import4.StaticNodeDebugInfo([
    import18.TemplateRef,
    import20.NgFor
  ]
  ,(null as any),{}),
  new import4.StaticNodeDebugInfo(([] as any[]),(null as any),{}),
  new import4.StaticNodeDebugInfo(([] as any[]),(null as any),{}),
  new import4.StaticNodeDebugInfo([
    import18.TemplateRef,
    import20.NgFor
  ]
  ,(null as any),{}),
  new import4.StaticNodeDebugInfo(([] as any[]),(null as any),{}),
  new import4.StaticNodeDebugInfo(([] as any[]),(null as any),{})
]
;
var renderType_DayCalendarComponent:import6.RenderComponentType = import3.createRenderComponentType('/Users/jonoforbes/Developer/Ether/winecellar/node_modules/ng2-date-picker/dp-day-calendar/dp-day-calendar.component.d.ts class DayCalendarComponent - inline template',0,import7.ViewEncapsulation.Emulated,styles_DayCalendarComponent,{});
export class View_DayCalendarComponent0 extends import2.DebugAppView<import0.DayCalendarComponent> {
  _el_0:any;
  _NgClass_0_3:import14.Wrapper_NgClass;
  _text_1:any;
  _el_2:any;
  _text_3:any;
  _anchor_4:any;
  /*private*/ _vc_4:import12.ViewContainer;
  _TemplateRef_4_5:any;
  _NgFor_4_6:import22.Wrapper_NgFor;
  _text_5:any;
  _text_6:any;
  _anchor_7:any;
  /*private*/ _vc_7:import12.ViewContainer;
  _TemplateRef_7_5:any;
  _NgFor_7_6:import22.Wrapper_NgFor;
  _text_8:any;
  _text_9:any;
  _map_17:any;
  constructor(viewUtils:import3.ViewUtils,parentView:import2.AppView<any>,parentIndex:number,parentElement:any) {
    super(View_DayCalendarComponent0,renderType_DayCalendarComponent,import8.ViewType.COMPONENT,viewUtils,parentView,parentIndex,parentElement,import9.ChangeDetectorStatus.CheckAlways,nodeDebugInfos_DayCalendarComponent0);
    this._map_17 = import3.pureProxy1((p0:any):{[key: string]:any} => {
      return {'dp-hide-near-month': p0};
    });
  }
  createInternal(rootSelector:string):import10.ComponentRef<any> {
    const parentRenderNode:any = this.renderer.createViewRoot(this.parentElement);
    this._el_0 = import3.createRenderElement(this.renderer,parentRenderNode,'div',new import3.InlineArray2(2,'class','dp-calendar-wrapper'),this.debug(0,0,0));
    this._NgClass_0_3 = new import14.Wrapper_NgClass(this.parentView.injectorGet(import15.IterableDiffers,this.parentIndex),this.parentView.injectorGet(import16.KeyValueDiffers,this.parentIndex),new import17.ElementRef(this._el_0),this.renderer);
    this._text_1 = this.renderer.createText(this._el_0,'   ',this.debug(1,0,95));
    this._el_2 = import3.createRenderElement(this.renderer,this._el_0,'div',new import3.InlineArray2(2,'class','dp-weekdays'),this.debug(2,0,98));
    this._text_3 = this.renderer.createText(this._el_2,'     ',this.debug(3,0,123));
    this._anchor_4 = this.renderer.createTemplateAnchor(this._el_2,this.debug(4,0,128));
    this._vc_4 = new import12.ViewContainer(4,2,this,this._anchor_4);
    this._TemplateRef_4_5 = new import18.TemplateRef_(this,4,this._anchor_4);
    this._NgFor_4_6 = new import22.Wrapper_NgFor(this._vc_4.vcRef,this._TemplateRef_4_5,this.parentView.injectorGet(import15.IterableDiffers,this.parentIndex),this.ref);
    this._text_5 = this.renderer.createText(this._el_2,'   ',this.debug(5,0,213));
    this._text_6 = this.renderer.createText(this._el_0,'   ',this.debug(6,0,222));
    this._anchor_7 = this.renderer.createTemplateAnchor(this._el_0,this.debug(7,0,225));
    this._vc_7 = new import12.ViewContainer(7,0,this,this._anchor_7);
    this._TemplateRef_7_5 = new import18.TemplateRef_(this,7,this._anchor_7);
    this._NgFor_7_6 = new import22.Wrapper_NgFor(this._vc_7.vcRef,this._TemplateRef_7_5,this.parentView.injectorGet(import15.IterableDiffers,this.parentIndex),this.ref);
    this._text_8 = this.renderer.createText(this._el_0,' ',this.debug(8,0,890));
    this._text_9 = this.renderer.createText(parentRenderNode,' ',this.debug(9,0,897));
    this.init((null as any),((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1,
      this._el_2,
      this._text_3,
      this._anchor_4,
      this._text_5,
      this._text_6,
      this._anchor_7,
      this._text_8,
      this._text_9
    ]
    ),(null as any));
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import18.TemplateRef) && (4 === requestNodeIndex))) { return this._TemplateRef_4_5; }
    if (((token === import20.NgFor) && (4 === requestNodeIndex))) { return this._NgFor_4_6.context; }
    if (((token === import18.TemplateRef) && (7 === requestNodeIndex))) { return this._TemplateRef_7_5; }
    if (((token === import20.NgFor) && (7 === requestNodeIndex))) { return this._NgFor_7_6.context; }
    if (((token === import13.NgClass) && ((0 <= requestNodeIndex) && (requestNodeIndex <= 8)))) { return this._NgClass_0_3.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    this.debug(0,0,5);
    const currVal_0_0_0:any = 'dp-calendar-wrapper';
    this._NgClass_0_3.check_klass(currVal_0_0_0,throwOnChange,false);
    this.debug(0,0,33);
    const currVal_0_0_1:any = this._map_17(!this.context.config.showNearMonthDays);
    this._NgClass_0_3.check_ngClass(currVal_0_0_1,throwOnChange,false);
    this._NgClass_0_3.ngDoCheck(this,this._el_0,throwOnChange);
    this.debug(4,0,162);
    const currVal_4_0_0:any = this.context.weekdays;
    this._NgFor_4_6.check_ngForOf(currVal_4_0_0,throwOnChange,false);
    this._NgFor_4_6.ngDoCheck(this,this._anchor_4,throwOnChange);
    this.debug(7,0,255);
    const currVal_7_0_0:any = this.context.weeks;
    this._NgFor_7_6.check_ngForOf(currVal_7_0_0,throwOnChange,false);
    this._NgFor_7_6.ngDoCheck(this,this._anchor_7,throwOnChange);
    this._vc_4.detectChangesInNestedViews(throwOnChange);
    this._vc_7.detectChangesInNestedViews(throwOnChange);
  }
  destroyInternal():void {
    this._vc_4.destroyNestedViews();
    this._vc_7.destroyNestedViews();
    this.debug(4,0,128);
    this.debug(7,0,225);
  }
  createEmbeddedViewInternal(nodeIndex:number):import2.AppView<any> {
    if ((nodeIndex == 4)) { return new View_DayCalendarComponent1(this.viewUtils,this,4,this._anchor_4,this._vc_4); }
    if ((nodeIndex == 7)) { return new View_DayCalendarComponent2(this.viewUtils,this,7,this._anchor_7,this._vc_7); }
    return (null as any);
  }
}
