// ng2-date-picker/dp-date-picker.module.js
"use strict";
var core_1 = require("@angular/core");
var forms_1 = require("@angular/forms");
var common_1 = require("@angular/common");
var dp_day_picker_component_1 = require("./dp-day-picker/dp-day-picker.component");
var dp_day_calendar_component_1 = require("./dp-day-calendar/dp-day-calendar.component");
var dp_calendar_component_1 = require("./dp-calendar/dp-calendar.component");
var dom_appender_service_1 = require("./common/services/dom-appender/dom-appender.service");
var dp_month_calendar_component_1 = require("./dp-month-calendar/dp-month-calendar.component");
var utils_service_1 = require("./common/services/utils/utils.service");
var dp_day_picker_component_2 = require("./dp-day-picker/dp-day-picker.component");
exports.DpDayPickerComponent = dp_day_picker_component_2.DpDayPickerComponent;
var dp_day_calendar_component_2 = require("./dp-day-calendar/dp-day-calendar.component");
exports.DayCalendarComponent = dp_day_calendar_component_2.DayCalendarComponent;
var dp_calendar_component_2 = require("./dp-calendar/dp-calendar.component");
exports.CalendarComponent = dp_calendar_component_2.CalendarComponent;
var DpDatePickerModule = (function () {
    function DpDatePickerModule() {
    }
    return DpDatePickerModule;
}());
DpDatePickerModule.decorators = [
    { type: core_1.NgModule, args: [{
                providers: [
                    dom_appender_service_1.DomHelper,
                    utils_service_1.UtilsService
                ],
                declarations: [
                    dp_day_picker_component_1.DpDayPickerComponent,
                    dp_day_calendar_component_1.DayCalendarComponent,
                    dp_calendar_component_1.CalendarComponent,
                    dp_month_calendar_component_1.MonthCalendarComponent
                ],
                imports: [
                    common_1.CommonModule,
                    forms_1.FormsModule
                ],
                exports: [dp_day_picker_component_1.DpDayPickerComponent, dp_day_calendar_component_1.DayCalendarComponent, dp_calendar_component_1.CalendarComponent]
            },] },
];
/** @nocollapse */
DpDatePickerModule.ctorParameters = function () { return []; };
exports.DpDatePickerModule = DpDatePickerModule;
