// ng2-date-picker/dp-calendar-month/dp-calendar-month.component.js
"use strict";
var core_1 = require('@angular/core');
var calendar_month_service_1 = require('./service/calendar-month.service');
var DpCalendarMonthComponent = (function () {
    function DpCalendarMonthComponent(calendarService) {
        this.calendarService = calendarService;
        this.dayClick = new core_1.EventEmitter();
        this.dayContextMenu = new core_1.EventEmitter();
    }
    DpCalendarMonthComponent.prototype.ngOnInit = function () {
        this.weeks = this.calendarService.generateMonthArray(this.config, this.selected);
        this.weekdays = this.calendarService.generateWeekdays(this.config.firstDayOfWeek, this.config.weekdayNames);
    };
    DpCalendarMonthComponent.prototype.ngOnChanges = function (changes) {
        var selected = changes.selected;
        if (selected && !selected.isFirstChange()) {
            this.weeks = this.calendarService
                .generateMonthArray(this.config, this.selected);
        }
    };
    DpCalendarMonthComponent.prototype.isDisabledDay = function (day) {
        return this.calendarService.isDateDisabled(day, this.config);
    };
    DpCalendarMonthComponent.decorators = [
        { type: core_1.Component, args: [{
                    selector: 'dp-calendar-month',
                    template: '<div class="dp-calendar-wrapper">   <div class="dp-weekdays">     <span class="dp-calendar-weekday" *ngFor="let weekday of weekdays">{{weekday}}</span>   </div>   <div class="dp-calendar-week" *ngFor="let week of weeks">     <button class="dp-calendar-day"             *ngFor="let day of week"             (click)="dayClick.emit({day: day, event: $event})"             [disabled]="isDisabledDay(day)"             [ngClass]="{               \'dp-selected\': day.selected,               \'dp-current-month\': day.currentMonth,               \'dp-prev-month\': day.prevMonth,               \'dp-next-month\': day.nextMonth,               \'dp-current-day\': day.currentDay             }">       {{day.date.format(\'DD\')}}     </button>   </div> </div> ',
                    styles: [':host {  display: inline-block;}.dp-calendar-wrapper {  box-sizing: border-box;  border: 1px solid #000000;}.dp-calendar-wrapper .dp-calendar-weekday:first-child {  border-left: none;}.dp-calendar-weekday {  box-sizing: border-box;  display: inline-block;  width: 30px;  text-align: center;  border-left: 1px solid #000000;  border-bottom: 1px solid #000000;}.dp-calendar-day {  box-sizing: border-box;  width: 30px;  height: 30px;  cursor: pointer;}.dp-selected {  background: blue;}.dp-prev-month,.dp-next-month {  opacity: 0.5;}:host(.dp-material) .dp-calendar-weekday {  height: 25px;  width: 30px;  line-height: 25px;  background: #E0E0E0;  border: none;}:host(.dp-material) .dp-calendar-wrapper {  border: none;}:host(.dp-material) .dp-calendar-day {  box-sizing: border-box;  height: 30px;  width: 30px;  background: white;  border-radius: 50%;  border: none;  outline: none;}:host(.dp-material) .dp-calendar-day:hover {  background: #E0E0E0;}:host(.dp-material) .dp-selected {  background: #106CC8;  color: white;}:host(.dp-material) .dp-selected:hover {  background: #106CC8;}:host(.dp-material) .dp-current-day {  border: 1px solid #106CC8;}'],
                    providers: [calendar_month_service_1.CalendarMonthService]
                },] },
    ];
    /** @nocollapse */
    DpCalendarMonthComponent.ctorParameters = function () { return [
        { type: calendar_month_service_1.CalendarMonthService, },
    ]; };
    DpCalendarMonthComponent.propDecorators = {
        'config': [{ type: core_1.Input },],
        'selected': [{ type: core_1.Input },],
        'dayClick': [{ type: core_1.Output },],
        'dayContextMenu': [{ type: core_1.Output },],
    };
    return DpCalendarMonthComponent;
}());
exports.DpCalendarMonthComponent = DpCalendarMonthComponent;
