// ng2-accordion/AccordionGroup.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular 2 template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */

import * as import0 from 'ng2-accordion/AccordionGroup';
import * as import1 from '@angular/core/src/change_detection/change_detection_util';
import * as import2 from '@angular/core/src/linker/view';
import * as import3 from '@angular/core/src/linker/view_utils';
import * as import4 from '@angular/core/src/linker/debug_context';
import * as import5 from '@angular/core/src/render/api';
import * as import6 from '@angular/core/src/metadata/view';
import * as import7 from '@angular/core/src/linker/query_list';
import * as import8 from '@angular/core/src/linker/view_type';
import * as import9 from '@angular/core/src/change_detection/constants';
import * as import10 from '@angular/core/src/linker/component_factory';
import * as import11 from 'ng2-accordion/Accordion';
import * as import12 from '@angular/core/src/linker/view_container';
import * as import13 from '@angular/core/src/linker/template_ref';
import * as import14 from '@angular/common/src/directives/ng_if';
import * as import15 from '../@angular/common/src/directives/ng_if.ngfactory';
import * as import16 from '@angular/core/src/security';
export class Wrapper_AccordionGroup {
  /*private*/ _eventHandler:Function;
  context:import0.AccordionGroup;
  /*private*/ _changed:boolean;
  /*private*/ _expr_0:any;
  /*private*/ _expr_1:any;
  /*private*/ _expr_2:any;
  subscription0:any;
  subscription1:any;
  subscription2:any;
  constructor(p0:any,p1:any) {
    this._changed = false;
    this.context = new import0.AccordionGroup(p0,p1);
    this._expr_0 = import1.UNINITIALIZED;
    this._expr_1 = import1.UNINITIALIZED;
    this._expr_2 = import1.UNINITIALIZED;
  }
  ngOnDetach(view:import2.AppView<any>,componentView:import2.AppView<any>,el:any):void {
  }
  ngOnDestroy():void {
    (this.subscription0 && this.subscription0.unsubscribe());
    (this.subscription1 && this.subscription1.unsubscribe());
    (this.subscription2 && this.subscription2.unsubscribe());
  }
  check_heading(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import3.checkBinding(throwOnChange,this._expr_0,currValue))) {
      this._changed = true;
      this.context.heading = currValue;
      this._expr_0 = currValue;
    }
  }
  check_isOpened(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import3.checkBinding(throwOnChange,this._expr_1,currValue))) {
      this._changed = true;
      this.context.isOpened = currValue;
      this._expr_1 = currValue;
    }
  }
  check_disabled(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import3.checkBinding(throwOnChange,this._expr_2,currValue))) {
      this._changed = true;
      this.context.disabled = currValue;
      this._expr_2 = currValue;
    }
  }
  ngDoCheck(view:import2.AppView<any>,el:any,throwOnChange:boolean):boolean {
    var changed:any = this._changed;
    this._changed = false;
    return changed;
  }
  checkHost(view:import2.AppView<any>,componentView:import2.AppView<any>,el:any,throwOnChange:boolean):void {
  }
  handleEvent(eventName:string,$event:any):boolean {
    var result:boolean = true;
    return result;
  }
  subscribe(view:import2.AppView<any>,_eventHandler:any,emit0:boolean,emit1:boolean,emit2:boolean):void {
    this._eventHandler = _eventHandler;
    if (emit0) { (this.subscription0 = this.context.onOpen.subscribe(_eventHandler.bind(view,'onOpen'))); }
    if (emit1) { (this.subscription1 = this.context.onClose.subscribe(_eventHandler.bind(view,'onClose'))); }
    if (emit2) { (this.subscription2 = this.context.onToggle.subscribe(_eventHandler.bind(view,'onToggle'))); }
  }
}
const nodeDebugInfos_AccordionGroup_Host0:import4.StaticNodeDebugInfo[] = [new import4.StaticNodeDebugInfo([import0.AccordionGroup],import0.AccordionGroup,{})];
var renderType_AccordionGroup_Host:import5.RenderComponentType = import3.createRenderComponentType('',0,import6.ViewEncapsulation.None,([] as any[]),{});
class View_AccordionGroup_Host0 extends import2.DebugAppView<any> {
  _el_0:any;
  compView_0:import2.AppView<import0.AccordionGroup>;
  _AccordionGroup_0_3:Wrapper_AccordionGroup;
  _query_AccordionToggle_0_0:import7.QueryList<any>;
  constructor(viewUtils:import3.ViewUtils,parentView:import2.AppView<any>,parentIndex:number,parentElement:any) {
    super(View_AccordionGroup_Host0,renderType_AccordionGroup_Host,import8.ViewType.HOST,viewUtils,parentView,parentIndex,parentElement,import9.ChangeDetectorStatus.CheckAlways,nodeDebugInfos_AccordionGroup_Host0);
  }
  createInternal(rootSelector:string):import10.ComponentRef<any> {
    this._el_0 = import3.selectOrCreateRenderHostElement(this.renderer,'accordion-group',import3.EMPTY_INLINE_ARRAY,rootSelector,this.debug(0,0,0));
    this.compView_0 = new View_AccordionGroup0(this.viewUtils,this,0,this._el_0);
    this._AccordionGroup_0_3 = new Wrapper_AccordionGroup(this.injectorGet(import11.Accordion,this.parentIndex),this.compView_0.ref);
    this._query_AccordionToggle_0_0 = new import7.QueryList<any>();
    this.compView_0.create(this._AccordionGroup_0_3.context);
    this._query_AccordionToggle_0_0.reset(([] as any[]));
    this._AccordionGroup_0_3.context.toggler = this._query_AccordionToggle_0_0.first;
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [this._el_0]),(null as any));
    return new import10.ComponentRef_<any>(0,this,this._el_0,this._AccordionGroup_0_3.context);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import0.AccordionGroup) && (0 === requestNodeIndex))) { return this._AccordionGroup_0_3.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    this.debug(0,0,0);
    this._AccordionGroup_0_3.ngDoCheck(this,this._el_0,throwOnChange);
    this.compView_0.internalDetectChanges(throwOnChange);
  }
  destroyInternal():void {
    this.compView_0.destroy();
    this._AccordionGroup_0_3.ngOnDestroy();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
  visitProjectableNodesInternal(nodeIndex:number,ngContentIndex:number,cb:any,ctx:any):void {
    if (((nodeIndex == 0) && (ngContentIndex == 0))) {  }
    if (((nodeIndex == 0) && (ngContentIndex == 1))) {  }
  }
}
export const AccordionGroupNgFactory:import10.ComponentFactory<import0.AccordionGroup> = new import10.ComponentFactory<import0.AccordionGroup>('accordion-group',View_AccordionGroup_Host0,import0.AccordionGroup);
const styles_AccordionGroup:any[] = ([] as any[]);
const nodeDebugInfos_AccordionGroup1:import4.StaticNodeDebugInfo[] = [
  new import4.StaticNodeDebugInfo(([] as any[]),(null as any),{}),
  new import4.StaticNodeDebugInfo(([] as any[]),(null as any),{})
]
;
class View_AccordionGroup1 extends import2.DebugAppView<any> {
  _el_0:any;
  _text_1:any;
  /*private*/ _expr_2:any;
  /*private*/ _expr_3:any;
  constructor(viewUtils:import3.ViewUtils,parentView:import2.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import12.ViewContainer) {
    super(View_AccordionGroup1,renderType_AccordionGroup,import8.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import9.ChangeDetectorStatus.CheckAlways,nodeDebugInfos_AccordionGroup1,declaredViewContainer);
    this._expr_2 = import1.UNINITIALIZED;
    this._expr_3 = import1.UNINITIALIZED;
  }
  createInternal(rootSelector:string):import10.ComponentRef<any> {
    this._el_0 = import3.createRenderElement(this.renderer,(null as any),'a',new import3.InlineArray4(4,'data-toggle','collapse','role','button'),this.debug(0,4,8));
    this._text_1 = this.renderer.createText(this._el_0,'',this.debug(1,4,96));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1
    ]
    ),(null as any));
    return (null as any);
  }
  detectChangesInternal(throwOnChange:boolean):void {
    this.debug(0,4,64);
    const currVal_2:any = this.parentView.context.isOpened;
    if (import3.checkBinding(throwOnChange,this._expr_2,currVal_2)) {
      this.renderer.setElementAttribute(this._el_0,'aria-expanded',((currVal_2 == null)? (null as any): currVal_2.toString()));
      this._expr_2 = currVal_2;
    }
    this.debug(1,4,96);
    const currVal_3:any = import3.inlineInterpolate(1,'\n            ',this.parentView.context.heading,'\n        ');
    if (import3.checkBinding(throwOnChange,this._expr_3,currVal_3)) {
      this.renderer.setText(this._text_1,currVal_3);
      this._expr_3 = currVal_3;
    }
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
}
const nodeDebugInfos_AccordionGroup2:import4.StaticNodeDebugInfo[] = [
  new import4.StaticNodeDebugInfo(([] as any[]),(null as any),{}),
  new import4.StaticNodeDebugInfo(([] as any[]),(null as any),{}),
  new import4.StaticNodeDebugInfo(([] as any[]),(null as any),{}),
  new import4.StaticNodeDebugInfo(([] as any[]),(null as any),{}),
  new import4.StaticNodeDebugInfo(([] as any[]),(null as any),{}),
  new import4.StaticNodeDebugInfo(([] as any[]),(null as any),{})
]
;
class View_AccordionGroup2 extends import2.DebugAppView<any> {
  _el_0:any;
  _text_1:any;
  _el_2:any;
  _text_3:any;
  _text_4:any;
  _text_5:any;
  /*private*/ _expr_6:any;
  constructor(viewUtils:import3.ViewUtils,parentView:import2.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import12.ViewContainer) {
    super(View_AccordionGroup2,renderType_AccordionGroup,import8.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import9.ChangeDetectorStatus.CheckAlways,nodeDebugInfos_AccordionGroup2,declaredViewContainer);
    this._expr_6 = import1.UNINITIALIZED;
  }
  createInternal(rootSelector:string):import10.ComponentRef<any> {
    this._el_0 = import3.createRenderElement(this.renderer,(null as any),'div',new import3.InlineArray4(4,'class','panel-collapse collapse in','role','tabpanel'),this.debug(0,12,4));
    this._text_1 = this.renderer.createText(this._el_0,'\n      ',this.debug(1,12,110));
    this._el_2 = import3.createRenderElement(this.renderer,this._el_0,'div',new import3.InlineArray2(2,'class','panel-body'),this.debug(2,13,6));
    this._text_3 = this.renderer.createText(this._el_2,'\n        ',this.debug(3,13,30));
    this.debug((null as any),14,8);
    this.projectNodes(this._el_2,1);
    this._text_4 = this.renderer.createText(this._el_2,'\n      ',this.debug(4,14,33));
    this._text_5 = this.renderer.createText(this._el_0,'\n    ',this.debug(5,15,12));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1,
      this._el_2,
      this._text_3,
      this._text_4,
      this._text_5
    ]
    ),(null as any));
    return (null as any);
  }
  detectChangesInternal(throwOnChange:boolean):void {
    this.debug(0,12,77);
    const currVal_6:any = this.parentView.context.heading;
    if (import3.checkBinding(throwOnChange,this._expr_6,currVal_6)) {
      this.renderer.setElementAttribute(this._el_0,'aria-labelledby',((currVal_6 == null)? (null as any): currVal_6.toString()));
      this._expr_6 = currVal_6;
    }
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
}
const nodeDebugInfos_AccordionGroup0:import4.StaticNodeDebugInfo[] = [
  new import4.StaticNodeDebugInfo(([] as any[]),(null as any),{}),
  new import4.StaticNodeDebugInfo(([] as any[]),(null as any),{}),
  new import4.StaticNodeDebugInfo(([] as any[]),(null as any),{}),
  new import4.StaticNodeDebugInfo(([] as any[]),(null as any),{}),
  new import4.StaticNodeDebugInfo(([] as any[]),(null as any),{}),
  new import4.StaticNodeDebugInfo(([] as any[]),(null as any),{}),
  new import4.StaticNodeDebugInfo(([] as any[]),(null as any),{}),
  new import4.StaticNodeDebugInfo([
    import13.TemplateRef,
    import14.NgIf
  ]
  ,(null as any),{}),
  new import4.StaticNodeDebugInfo(([] as any[]),(null as any),{}),
  new import4.StaticNodeDebugInfo(([] as any[]),(null as any),{}),
  new import4.StaticNodeDebugInfo(([] as any[]),(null as any),{}),
  new import4.StaticNodeDebugInfo(([] as any[]),(null as any),{}),
  new import4.StaticNodeDebugInfo(([] as any[]),(null as any),{}),
  new import4.StaticNodeDebugInfo(([] as any[]),(null as any),{}),
  new import4.StaticNodeDebugInfo(([] as any[]),(null as any),{}),
  new import4.StaticNodeDebugInfo([
    import13.TemplateRef,
    import14.NgIf
  ]
  ,(null as any),{}),
  new import4.StaticNodeDebugInfo(([] as any[]),(null as any),{}),
  new import4.StaticNodeDebugInfo(([] as any[]),(null as any),{})
]
;
var renderType_AccordionGroup:import5.RenderComponentType = import3.createRenderComponentType('/Users/jonoforbes/Developer/Ether/winecellar/node_modules/ng2-accordion/AccordionGroup.d.ts class AccordionGroup - inline template',2,import6.ViewEncapsulation.None,styles_AccordionGroup,{});
export class View_AccordionGroup0 extends import2.DebugAppView<import0.AccordionGroup> {
  _text_0:any;
  _el_1:any;
  _text_2:any;
  _el_3:any;
  _text_4:any;
  _el_5:any;
  _text_6:any;
  _anchor_7:any;
  /*private*/ _vc_7:import12.ViewContainer;
  _TemplateRef_7_5:any;
  _NgIf_7_6:import15.Wrapper_NgIf;
  _text_8:any;
  _text_9:any;
  _el_10:any;
  _text_11:any;
  _text_12:any;
  _text_13:any;
  _text_14:any;
  _anchor_15:any;
  /*private*/ _vc_15:import12.ViewContainer;
  _TemplateRef_15_5:any;
  _NgIf_15_6:import15.Wrapper_NgIf;
  _text_16:any;
  _text_17:any;
  /*private*/ _expr_24:any;
  /*private*/ _expr_25:any;
  /*private*/ _expr_26:any;
  constructor(viewUtils:import3.ViewUtils,parentView:import2.AppView<any>,parentIndex:number,parentElement:any) {
    super(View_AccordionGroup0,renderType_AccordionGroup,import8.ViewType.COMPONENT,viewUtils,parentView,parentIndex,parentElement,import9.ChangeDetectorStatus.CheckAlways,nodeDebugInfos_AccordionGroup0);
    this._expr_24 = import1.UNINITIALIZED;
    this._expr_25 = import1.UNINITIALIZED;
    this._expr_26 = import1.UNINITIALIZED;
  }
  createInternal(rootSelector:string):import10.ComponentRef<any> {
    const parentRenderNode:any = this.renderer.createViewRoot(this.parentElement);
    this._text_0 = this.renderer.createText(parentRenderNode,'\n  ',this.debug(0,0,0));
    this._el_1 = import3.createRenderElement(this.renderer,parentRenderNode,'div',new import3.InlineArray2(2,'class','panel panel-default'),this.debug(1,1,2));
    this._text_2 = this.renderer.createText(this._el_1,'\n    ',this.debug(2,1,89));
    this._el_3 = import3.createRenderElement(this.renderer,this._el_1,'div',new import3.InlineArray4(4,'class','panel-heading','role','tab'),this.debug(3,2,4));
    this._text_4 = this.renderer.createText(this._el_3,'\n      ',this.debug(4,2,69));
    this._el_5 = import3.createRenderElement(this.renderer,this._el_3,'h4',new import3.InlineArray2(2,'class','panel-title'),this.debug(5,3,6));
    this._text_6 = this.renderer.createText(this._el_5,'\n        ',this.debug(6,3,30));
    this._anchor_7 = this.renderer.createTemplateAnchor(this._el_5,this.debug(7,4,8));
    this._vc_7 = new import12.ViewContainer(7,5,this,this._anchor_7);
    this._TemplateRef_7_5 = new import13.TemplateRef_(this,7,this._anchor_7);
    this._NgIf_7_6 = new import15.Wrapper_NgIf(this._vc_7.vcRef,this._TemplateRef_7_5);
    this._text_8 = this.renderer.createText(this._el_5,'\n        ',this.debug(8,6,12));
    this.debug((null as any),7,8);
    this.projectNodes(this._el_5,0);
    this._text_9 = this.renderer.createText(this._el_5,'\n        ',this.debug(9,7,60));
    this._el_10 = import3.createRenderElement(this.renderer,this._el_5,'div',new import3.InlineArray2(2,'class','caret'),this.debug(10,8,8));
    this._text_11 = this.renderer.createText(this._el_10,'\n        ',this.debug(11,8,80));
    this._text_12 = this.renderer.createText(this._el_5,'\n      ',this.debug(12,9,14));
    this._text_13 = this.renderer.createText(this._el_3,'\n    ',this.debug(13,10,11));
    this._text_14 = this.renderer.createText(this._el_1,'\n    ',this.debug(14,11,10));
    this._anchor_15 = this.renderer.createTemplateAnchor(this._el_1,this.debug(15,12,4));
    this._vc_15 = new import12.ViewContainer(15,1,this,this._anchor_15);
    this._TemplateRef_15_5 = new import13.TemplateRef_(this,15,this._anchor_15);
    this._NgIf_15_6 = new import15.Wrapper_NgIf(this._vc_15.vcRef,this._TemplateRef_15_5);
    this._text_16 = this.renderer.createText(this._el_1,'\n  ',this.debug(16,16,10));
    this._text_17 = this.renderer.createText(parentRenderNode,'\n',this.debug(17,17,8));
    var disposable_0:Function = import3.subscribeToRenderElement(this,this._el_3,new import3.InlineArray2(2,'click',(null as any)),this.eventHandler(this.handleEvent_3));
    this.init((null as any),((<any>this.renderer).directRenderer? (null as any): [
      this._text_0,
      this._el_1,
      this._text_2,
      this._el_3,
      this._text_4,
      this._el_5,
      this._text_6,
      this._anchor_7,
      this._text_8,
      this._text_9,
      this._el_10,
      this._text_11,
      this._text_12,
      this._text_13,
      this._text_14,
      this._anchor_15,
      this._text_16,
      this._text_17
    ]
    ),[disposable_0]);
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import13.TemplateRef) && (7 === requestNodeIndex))) { return this._TemplateRef_7_5; }
    if (((token === import14.NgIf) && (7 === requestNodeIndex))) { return this._NgIf_7_6.context; }
    if (((token === import13.TemplateRef) && (15 === requestNodeIndex))) { return this._TemplateRef_15_5; }
    if (((token === import14.NgIf) && (15 === requestNodeIndex))) { return this._NgIf_15_6.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    this.debug(7,4,11);
    const currVal_7_0_0:any = this.context.heading;
    this._NgIf_7_6.check_ngIf(currVal_7_0_0,throwOnChange,false);
    this._NgIf_7_6.ngDoCheck(this,this._anchor_7,throwOnChange);
    this.debug(15,12,9);
    const currVal_15_0_0:any = this.context.isOpened;
    this._NgIf_15_6.check_ngIf(currVal_15_0_0,throwOnChange,false);
    this._NgIf_15_6.ngDoCheck(this,this._anchor_15,throwOnChange);
    this._vc_7.detectChangesInNestedViews(throwOnChange);
    this._vc_15.detectChangesInNestedViews(throwOnChange);
    this.debug(1,1,35);
    const currVal_24:any = this.context.isOpened;
    if (import3.checkBinding(throwOnChange,this._expr_24,currVal_24)) {
      this.renderer.setElementClass(this._el_1,'dropup',currVal_24);
      this._expr_24 = currVal_24;
    }
    this.debug(1,1,61);
    const currVal_25:any = this.context.disabled;
    if (import3.checkBinding(throwOnChange,this._expr_25,currVal_25)) {
      this.renderer.setElementClass(this._el_1,'disabled',currVal_25);
      this._expr_25 = currVal_25;
    }
    this.debug(10,8,27);
    const currVal_26:any = (this.context.accordion.showArrows? '': 'none');
    if (import3.checkBinding(throwOnChange,this._expr_26,currVal_26)) {
      this.renderer.setElementStyle(this._el_10,'display',((this.viewUtils.sanitizer.sanitize(import16.SecurityContext.STYLE,currVal_26) == null)? (null as any): this.viewUtils.sanitizer.sanitize(import16.SecurityContext.STYLE,currVal_26).toString()));
      this._expr_26 = currVal_26;
    }
  }
  destroyInternal():void {
    this._vc_7.destroyNestedViews();
    this._vc_15.destroyNestedViews();
    this.debug(7,4,8);
  }
  createEmbeddedViewInternal(nodeIndex:number):import2.AppView<any> {
    if ((nodeIndex == 7)) { return new View_AccordionGroup1(this.viewUtils,this,7,this._anchor_7,this._vc_7); }
    if ((nodeIndex == 15)) { return new View_AccordionGroup2(this.viewUtils,this,15,this._anchor_15,this._vc_15); }
    return (null as any);
  }
  handleEvent_3(eventName:string,$event:any):boolean {
    this.debug(3,2,4);
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    if ((eventName == 'click')) {
      const pd_sub_0:any = ((<any>this.context.checkAndToggle()) !== false);
      result = (pd_sub_0 && result);
    }
    return result;
  }
}
