// ngx-popover/PopoverContent.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular 2 template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */

import * as import0 from 'ngx-popover/PopoverContent';
import * as import1 from '@angular/core/src/change_detection/change_detection_util';
import * as import2 from '@angular/core/src/linker/view';
import * as import3 from '@angular/core/src/linker/view_utils';
import * as import4 from '@angular/core/src/linker/debug_context';
import * as import5 from '@angular/core/src/render/api';
import * as import6 from '@angular/core/src/metadata/view';
import * as import7 from '@angular/core/src/linker/view_type';
import * as import8 from '@angular/core/src/change_detection/constants';
import * as import9 from '@angular/core/src/linker/component_factory';
import * as import10 from '@angular/core/src/linker/element_ref';
import * as import11 from '@angular/core/src/linker/query_list';
import * as import12 from '@angular/core/src/security';
export class Wrapper_PopoverContent {
  /*private*/ _eventHandler:Function;
  context:import0.PopoverContent;
  /*private*/ _changed:boolean;
  /*private*/ _expr_0:any;
  /*private*/ _expr_1:any;
  /*private*/ _expr_2:any;
  /*private*/ _expr_3:any;
  /*private*/ _expr_4:any;
  /*private*/ _expr_5:any;
  constructor(p0:any,p1:any,p2:any) {
    this._changed = false;
    this.context = new import0.PopoverContent(p0,p1,p2);
    this._expr_0 = import1.UNINITIALIZED;
    this._expr_1 = import1.UNINITIALIZED;
    this._expr_2 = import1.UNINITIALIZED;
    this._expr_3 = import1.UNINITIALIZED;
    this._expr_4 = import1.UNINITIALIZED;
    this._expr_5 = import1.UNINITIALIZED;
  }
  ngOnDetach(view:import2.AppView<any>,componentView:import2.AppView<any>,el:any):void {
  }
  ngOnDestroy():void {
    this.context.ngOnDestroy();
  }
  check_content(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import3.checkBinding(throwOnChange,this._expr_0,currValue))) {
      this._changed = true;
      this.context.content = currValue;
      this._expr_0 = currValue;
    }
  }
  check_placement(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import3.checkBinding(throwOnChange,this._expr_1,currValue))) {
      this._changed = true;
      this.context.placement = currValue;
      this._expr_1 = currValue;
    }
  }
  check_title(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import3.checkBinding(throwOnChange,this._expr_2,currValue))) {
      this._changed = true;
      this.context.title = currValue;
      this._expr_2 = currValue;
    }
  }
  check_animation(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import3.checkBinding(throwOnChange,this._expr_3,currValue))) {
      this._changed = true;
      this.context.animation = currValue;
      this._expr_3 = currValue;
    }
  }
  check_closeOnClickOutside(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import3.checkBinding(throwOnChange,this._expr_4,currValue))) {
      this._changed = true;
      this.context.closeOnClickOutside = currValue;
      this._expr_4 = currValue;
    }
  }
  check_closeOnMouseOutside(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import3.checkBinding(throwOnChange,this._expr_5,currValue))) {
      this._changed = true;
      this.context.closeOnMouseOutside = currValue;
      this._expr_5 = currValue;
    }
  }
  ngDoCheck(view:import2.AppView<any>,el:any,throwOnChange:boolean):boolean {
    var changed:any = this._changed;
    this._changed = false;
    return changed;
  }
  checkHost(view:import2.AppView<any>,componentView:import2.AppView<any>,el:any,throwOnChange:boolean):void {
  }
  handleEvent(eventName:string,$event:any):boolean {
    var result:boolean = true;
    return result;
  }
  subscribe(view:import2.AppView<any>,_eventHandler:any):void {
    this._eventHandler = _eventHandler;
  }
}
const nodeDebugInfos_PopoverContent_Host0:import4.StaticNodeDebugInfo[] = [new import4.StaticNodeDebugInfo([import0.PopoverContent],import0.PopoverContent,{})];
var renderType_PopoverContent_Host:import5.RenderComponentType = import3.createRenderComponentType('',0,import6.ViewEncapsulation.None,([] as any[]),{});
class View_PopoverContent_Host0 extends import2.DebugAppView<any> {
  _el_0:any;
  compView_0:import2.AppView<import0.PopoverContent>;
  _PopoverContent_0_3:Wrapper_PopoverContent;
  constructor(viewUtils:import3.ViewUtils,parentView:import2.AppView<any>,parentIndex:number,parentElement:any) {
    super(View_PopoverContent_Host0,renderType_PopoverContent_Host,import7.ViewType.HOST,viewUtils,parentView,parentIndex,parentElement,import8.ChangeDetectorStatus.CheckAlways,nodeDebugInfos_PopoverContent_Host0);
  }
  createInternal(rootSelector:string):import9.ComponentRef<any> {
    this._el_0 = import3.selectOrCreateRenderHostElement(this.renderer,'popover-content',import3.EMPTY_INLINE_ARRAY,rootSelector,this.debug(0,0,0));
    this.compView_0 = new View_PopoverContent0(this.viewUtils,this,0,this._el_0);
    this._PopoverContent_0_3 = new Wrapper_PopoverContent(new import10.ElementRef(this._el_0),this.compView_0.ref,this.renderer);
    this.compView_0.create(this._PopoverContent_0_3.context);
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [this._el_0]),(null as any));
    return new import9.ComponentRef_<any>(0,this,this._el_0,this._PopoverContent_0_3.context);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import0.PopoverContent) && (0 === requestNodeIndex))) { return this._PopoverContent_0_3.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    this.debug(0,0,0);
    this._PopoverContent_0_3.ngDoCheck(this,this._el_0,throwOnChange);
    this.compView_0.internalDetectChanges(throwOnChange);
    if (!throwOnChange) {
      this.debug(0,0,0);
      if ((this.numberOfChecks === 0)) { this._PopoverContent_0_3.context.ngAfterViewInit(); }
    }
  }
  destroyInternal():void {
    this.compView_0.destroy();
    this._PopoverContent_0_3.ngOnDestroy();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
  visitProjectableNodesInternal(nodeIndex:number,ngContentIndex:number,cb:any,ctx:any):void {
    if (((nodeIndex == 0) && (ngContentIndex == 0))) {  }
  }
}
export const PopoverContentNgFactory:import9.ComponentFactory<import0.PopoverContent> = new import9.ComponentFactory<import0.PopoverContent>('popover-content',View_PopoverContent_Host0,import0.PopoverContent);
const styles_PopoverContent:any[] = ['.popover[_ngcontent-%COMP%]   .virtual-area[_ngcontent-%COMP%] {\n    height: 11px;\n    width: 100%;\n    position: absolute;\n}\n.popover.top[_ngcontent-%COMP%]   .virtual-area[_ngcontent-%COMP%] {\n    bottom: -11px; \n}\n.popover.bottom[_ngcontent-%COMP%]   .virtual-area[_ngcontent-%COMP%] {\n    top: -11px; \n}\n.popover.left[_ngcontent-%COMP%]   .virtual-area[_ngcontent-%COMP%] {\n    right: -11px; \n}\n.popover.right[_ngcontent-%COMP%]   .virtual-area[_ngcontent-%COMP%] {\n    left: -11px; \n}'];
const nodeDebugInfos_PopoverContent0:import4.StaticNodeDebugInfo[] = [
  new import4.StaticNodeDebugInfo(([] as any[]),(null as any),{}),
  new import4.StaticNodeDebugInfo(([] as any[]),(null as any),{popoverDiv: (null as any)}),
  new import4.StaticNodeDebugInfo(([] as any[]),(null as any),{}),
  new import4.StaticNodeDebugInfo(([] as any[]),(null as any),{}),
  new import4.StaticNodeDebugInfo(([] as any[]),(null as any),{}),
  new import4.StaticNodeDebugInfo(([] as any[]),(null as any),{}),
  new import4.StaticNodeDebugInfo(([] as any[]),(null as any),{}),
  new import4.StaticNodeDebugInfo(([] as any[]),(null as any),{}),
  new import4.StaticNodeDebugInfo(([] as any[]),(null as any),{}),
  new import4.StaticNodeDebugInfo(([] as any[]),(null as any),{}),
  new import4.StaticNodeDebugInfo(([] as any[]),(null as any),{}),
  new import4.StaticNodeDebugInfo(([] as any[]),(null as any),{}),
  new import4.StaticNodeDebugInfo(([] as any[]),(null as any),{}),
  new import4.StaticNodeDebugInfo(([] as any[]),(null as any),{}),
  new import4.StaticNodeDebugInfo(([] as any[]),(null as any),{})
]
;
var renderType_PopoverContent:import5.RenderComponentType = import3.createRenderComponentType('/Users/jonoforbes/Developer/Ether/winecellar/node_modules/ngx-popover/PopoverContent.d.ts class PopoverContent - inline template',1,import6.ViewEncapsulation.Emulated,styles_PopoverContent,{});
export class View_PopoverContent0 extends import2.DebugAppView<import0.PopoverContent> {
  _viewQuery_popoverDiv_0:import11.QueryList<any>;
  _text_0:any;
  _el_1:any;
  _text_2:any;
  _el_3:any;
  _text_4:any;
  _el_5:any;
  _text_6:any;
  _el_7:any;
  _text_8:any;
  _text_9:any;
  _el_10:any;
  _text_11:any;
  _text_12:any;
  _text_13:any;
  _text_14:any;
  /*private*/ _expr_16:any;
  /*private*/ _expr_17:any;
  /*private*/ _expr_18:any;
  /*private*/ _expr_19:any;
  /*private*/ _expr_20:any;
  /*private*/ _expr_21:any;
  /*private*/ _expr_22:any;
  /*private*/ _expr_23:any;
  /*private*/ _expr_24:any;
  constructor(viewUtils:import3.ViewUtils,parentView:import2.AppView<any>,parentIndex:number,parentElement:any) {
    super(View_PopoverContent0,renderType_PopoverContent,import7.ViewType.COMPONENT,viewUtils,parentView,parentIndex,parentElement,import8.ChangeDetectorStatus.CheckAlways,nodeDebugInfos_PopoverContent0);
    this._expr_16 = import1.UNINITIALIZED;
    this._expr_17 = import1.UNINITIALIZED;
    this._expr_18 = import1.UNINITIALIZED;
    this._expr_19 = import1.UNINITIALIZED;
    this._expr_20 = import1.UNINITIALIZED;
    this._expr_21 = import1.UNINITIALIZED;
    this._expr_22 = import1.UNINITIALIZED;
    this._expr_23 = import1.UNINITIALIZED;
    this._expr_24 = import1.UNINITIALIZED;
  }
  createInternal(rootSelector:string):import9.ComponentRef<any> {
    const parentRenderNode:any = this.renderer.createViewRoot(this.parentElement);
    this._viewQuery_popoverDiv_0 = new import11.QueryList<any>();
    this._text_0 = this.renderer.createText(parentRenderNode,'\n',this.debug(0,0,0));
    this._el_1 = import3.createRenderElement(this.renderer,parentRenderNode,'div',new import3.InlineArray4(4,'role','popover','style','display: block'),this.debug(1,1,0));
    this._text_2 = this.renderer.createText(this._el_1,'\n    ',this.debug(2,7,20));
    this._el_3 = import3.createRenderElement(this.renderer,this._el_1,'div',new import3.InlineArray2(2,'class','virtual-area'),this.debug(3,8,4));
    this._text_4 = this.renderer.createText(this._el_1,'\n    ',this.debug(4,8,68));
    this._el_5 = import3.createRenderElement(this.renderer,this._el_1,'div',new import3.InlineArray2(2,'class','arrow'),this.debug(5,9,4));
    this._text_6 = this.renderer.createText(this._el_1,' \n    ',this.debug(6,9,29));
    this._el_7 = import3.createRenderElement(this.renderer,this._el_1,'h3',new import3.InlineArray2(2,'class','popover-title'),this.debug(7,10,4));
    this._text_8 = this.renderer.createText(this._el_7,'',this.debug(8,10,48));
    this._text_9 = this.renderer.createText(this._el_1,'\n    ',this.debug(9,10,64));
    this._el_10 = import3.createRenderElement(this.renderer,this._el_1,'div',new import3.InlineArray2(2,'class','popover-content'),this.debug(10,11,4));
    this._text_11 = this.renderer.createText(this._el_10,'\n        ',this.debug(11,11,33));
    this.debug((null as any),12,8);
    this.projectNodes(this._el_10,0);
    this._text_12 = this.renderer.createText(this._el_10,'',this.debug(12,12,33));
    this._text_13 = this.renderer.createText(this._el_1,' \n',this.debug(13,14,10));
    this._text_14 = this.renderer.createText(parentRenderNode,'\n',this.debug(14,15,6));
    this._viewQuery_popoverDiv_0.reset([new import10.ElementRef(this._el_1)]);
    this.context.popoverDiv = this._viewQuery_popoverDiv_0.first;
    this.init((null as any),((<any>this.renderer).directRenderer? (null as any): [
      this._text_0,
      this._el_1,
      this._text_2,
      this._el_3,
      this._text_4,
      this._el_5,
      this._text_6,
      this._el_7,
      this._text_8,
      this._text_9,
      this._el_10,
      this._text_11,
      this._text_12,
      this._text_13,
      this._text_14
    ]
    ),(null as any));
    return (null as any);
  }
  detectChangesInternal(throwOnChange:boolean):void {
    this.debug(1,1,17);
    const currVal_16:any = import3.inlineInterpolate(1,'popover ',this.context.effectivePlacement,'');
    if (import3.checkBinding(throwOnChange,this._expr_16,currVal_16)) {
      this.renderer.setElementProperty(this._el_1,'className',currVal_16);
      this._expr_16 = currVal_16;
    }
    this.debug(1,2,5);
    const currVal_17:any = (this.context.top + 'px');
    if (import3.checkBinding(throwOnChange,this._expr_17,currVal_17)) {
      this.renderer.setElementStyle(this._el_1,'top',((this.viewUtils.sanitizer.sanitize(import12.SecurityContext.STYLE,currVal_17) == null)? (null as any): this.viewUtils.sanitizer.sanitize(import12.SecurityContext.STYLE,currVal_17).toString()));
      this._expr_17 = currVal_17;
    }
    this.debug(1,3,5);
    const currVal_18:any = (this.context.left + 'px');
    if (import3.checkBinding(throwOnChange,this._expr_18,currVal_18)) {
      this.renderer.setElementStyle(this._el_1,'left',((this.viewUtils.sanitizer.sanitize(import12.SecurityContext.STYLE,currVal_18) == null)? (null as any): this.viewUtils.sanitizer.sanitize(import12.SecurityContext.STYLE,currVal_18).toString()));
      this._expr_18 = currVal_18;
    }
    this.debug(1,4,5);
    const currVal_19:any = this.context.isIn;
    if (import3.checkBinding(throwOnChange,this._expr_19,currVal_19)) {
      this.renderer.setElementClass(this._el_1,'in',currVal_19);
      this._expr_19 = currVal_19;
    }
    this.debug(1,5,5);
    const currVal_20:any = this.context.animation;
    if (import3.checkBinding(throwOnChange,this._expr_20,currVal_20)) {
      this.renderer.setElementClass(this._el_1,'fade',currVal_20);
      this._expr_20 = currVal_20;
    }
    this.debug(3,8,9);
    const currVal_21:boolean = !this.context.closeOnMouseOutside;
    if (import3.checkBinding(throwOnChange,this._expr_21,currVal_21)) {
      this.renderer.setElementProperty(this._el_3,'hidden',currVal_21);
      this._expr_21 = currVal_21;
    }
    this.debug(7,10,30);
    const currVal_22:boolean = !this.context.title;
    if (import3.checkBinding(throwOnChange,this._expr_22,currVal_22)) {
      this.renderer.setElementProperty(this._el_7,'hidden',currVal_22);
      this._expr_22 = currVal_22;
    }
    this.debug(8,10,48);
    const currVal_23:any = import3.inlineInterpolate(1,'',this.context.title,'');
    if (import3.checkBinding(throwOnChange,this._expr_23,currVal_23)) {
      this.renderer.setText(this._text_8,currVal_23);
      this._expr_23 = currVal_23;
    }
    this.debug(12,12,33);
    const currVal_24:any = import3.inlineInterpolate(1,'\n        ',this.context.content,'\n    ');
    if (import3.checkBinding(throwOnChange,this._expr_24,currVal_24)) {
      this.renderer.setText(this._text_12,currVal_24);
      this._expr_24 = currVal_24;
    }
  }
}
