// ngrx-undo/src/storeUndo.js
import { Reducer, Dispatcher, INITIAL_REDUCER, StoreModule } from "@ngrx/store";
import { OpaqueToken, NgModule } from "@angular/core";
import { createReducer } from "./createReducer";
export var UNDO_DEVTOOLS_CONFIG = new OpaqueToken('@ngrx/undodevtools Options');
export var UNDO_ACTION = "ngrx-undo/UNDO_ACTION";
export var StoreUndoModule = (function () {
    function StoreUndoModule() {
    }
    StoreUndoModule.interceptStore = function (options) {
        if (options === void 0) { options = { bufferSize: 100 }; }
        return {
            ngModule: StoreUndoModule,
            providers: [
                { provide: UNDO_DEVTOOLS_CONFIG, useValue: options },
                {
                    provide: Reducer,
                    deps: [Dispatcher, INITIAL_REDUCER, UNDO_DEVTOOLS_CONFIG],
                    useFactory: createReducer
                }
            ]
        };
    };
    StoreUndoModule.decorators = [
        { type: NgModule, args: [{
                    imports: [
                        StoreModule
                    ]
                },] },
    ];
    /** @nocollapse */
    StoreUndoModule.ctorParameters = [];
    return StoreUndoModule;
}());
